packages/commands/src/upload-track.mjs

```diff
@@ -60,6 +60,50 @@ const getPremiumConditions = async ({
   return null
 }
 
+const getDownloadConditions = async ({
+  downloadConditions,
+  downloadPrice: downloadPriceString,
+  audiusLibs
+}) => {
+  if (downloadPriceString) {
+    const price = Number.parseInt(downloadPriceString)
+    if (!Number.isFinite(price) || price <= 0) {
+      throw new Error(`Invalid price "${downloadPriceString}"`)
+    }
+    const { userbank } =
+      await audiusLibs.solanaWeb3Manager.createUserBankIfNeeded({
+        mint: 'usdc'
+      })
+    return {
+      usdc_purchase: {
+        price,
+        splits: { [userbank.toString()]: price * 10 ** 4 }
+      }
+    }
+  } else if (downloadConditions) {
+    return JSON.parse(downloadConditions)
+  }
+  return null
+}
+
+const getDownloadMetadata = (downloadConditions) => {
+  if (downloadConditions) {
+    return {
+      is_download_gated: true,
+      download_conditions: downloadConditions,
+      download: {
+        cid: '',
+        is_downloadable: true,
```

Comment:
maybe we should rethink about this old pre-existing download field in the track metadata. now that tracks may be download-gated, and that we expose both 320 cid and original cid, maybe this is no longer needed.
even the requires_follow just ends up being in the download_conditions if we were to remove this.

made a [ticket](https://linear.app/audius/issue/PAY-2260/rethink-the-old-download-field-in-track-metadata)

Reply:
Yeah I tend to agree... Also, is the existence of `download_conditions` enough to infer `is_download_gated`? Do we need separate fields? (I know this matches `is_premium` etc but maybe we can rethink that?)

Reply:
> Do we need separate fields? (I know this matches `is_premium` etc but maybe we can rethink that?)

This might be partially my influence. I'm not a big fan of inferred state. Though in this instance maybe it's not a big deal.

Reply:
sorry getting back to this so late. im down to do the following remove is_stream_gated and is_download_gated from track model, yet add them in the dn track metadata response that goes to client. this way, client will still have easy booleans to use. wdyt?